esi/src/parse.rs

```diff
@@ -61,28 +82,55 @@ impl TagNames {
             r#try: format!("{namespace}:try",).into_bytes(),
             attempt: format!("{namespace}:attempt",).into_bytes(),
             except: format!("{namespace}:except",).into_bytes(),
+            assign: format!("{namespace}:assign",).into_bytes(),
+            vars: format!("{namespace}:vars",).into_bytes(),
+            choose: format!("{namespace}:choose",).into_bytes(),
+            when: format!("{namespace}:when",).into_bytes(),
+            otherwise: format!("{namespace}:otherwise",).into_bytes(),
         }
     }
 }
 
+#[derive(Debug, PartialEq)]
+enum ContentType {
+    Normal,
+    Interpolated,
+}
+
 fn do_parse<'a, R>(
     reader: &mut Reader<R>,
     callback: &mut dyn FnMut(Event<'a>) -> Result<()>,
     task: &mut Vec<Event<'a>>,
-    depth: &mut usize,
+    use_queue: bool,
+    try_depth: &mut usize,
+    choose_depth: &mut usize,
     current_arm: &mut Option<TryTagArms>,
     tag: &TagNames,
+    content_type: &ContentType,
 ) -> Result<()>
 where
```

Comment:
This function signature is getting pretty complex but we can leave it for now as it's not part of the public API surface and there is a parser rework in progress.